server/catgenome/src/main/java/com/epam/catgenome/entity/FeatureFile.java

```diff
@@ -81,4 +84,16 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return getId().hashCode();
     }
+
+    public String getExtension() {
+        String[] pathParts = getPath().split("\\.");
+        if (getCompressed() && getPath().endsWith("gz") && pathParts.length > 2) {
+            return String.format("%s.%s", pathParts[pathParts.length - 2], pathParts[pathParts.length - 1]);
+        }
+        return pathParts[pathParts.length - 1];
+    }
+
+    public AsciiFeatureCodec<? extends Feature> getCodec() {
```

Comment:
As this method is used only for BedFile can we move it to this class or build codec `FileManager.makeBedReader` method?